nrf52-hal-common/src/lib.rs

```diff
@@ -83,9 +79,10 @@ impl DmaSlice {
 
 pub use crate::clocks::Clocks;
 pub use crate::delay::Delay;
+pub use crate::rng::Rng;
 pub use crate::rtc::Rtc;
 pub use crate::saadc::Saadc;
 pub use crate::spim::Spim;
-pub use crate::timer::Timer;
+pub use crate::timer::{Instance, Timer};
```

Comment:
None of the other re-exports pull in symbols from their modules. Is that desirable?

Reply:
Not sure, but the Instance is needed to have a generic function taking `Timer<T: Instance>`.
If there is a better fix we can do that, I just followed the error-train and ended up here :)

Reply:
It would be best if @hannobraun can give his thought to this as he is restructuring things currently and I am not 100% in the loop how he's laying out things.

Reply:
`Instance` shouldn't be re-exported, as that would conflict with the `Instance` traits of other peripherals. The intended use is `timer::Instance`.

Of course, we could call it `TimerInstance` and re-export it, or re-export it with `timer::Instance as TimerInstance`. I prefer using it as `timer::Instance`, but that's a matter of taste.

Reply:
I tested using `timer::Instance`, and this did not compile (it cannot find it). I then wonder what it is causing the error.

Reply:
I will test some more, hold this PR for now

Reply:
Hmm, if I remove the export (testing in my `rust-dwm1001` PR), the error becomes:

```
error[E0405]: cannot find trait `Instance` in module `timer`
   --> src/lib.rs:823:19
    |
823 |         T: timer::Instance,
    |                   ^^^^^^^^ not found in `timer`

error: aborting due to previous error
```
And `lib.rs` has `use nrf52832_hal::timer`, which should be enough, so something is wrong in its exporting.
Do you have any idea @hannobraun ?

Reply:
@korken Your downstream PR depends on version `0.8.1`, which doesn't include my changes. Those are not released yet.

Reply:
Yeah, you were correct. Fixed in force push!